pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -46,76 +46,81 @@ const (
 	netDeviceTypeVF   = "vf"
 )
 
+// Declared variables for test
+var (
+	ipSetupVethWithName            = cniip.SetupVethWithName
+	ipamConfigureIface             = ipam.ConfigureIface
+	ethtoolTXHWCsumOff             = ethtool.EthtoolTXHWCsumOff
+	renameInterface                = util.RenameInterface
+	netInterfaceByName             = net.InterfaceByName
+	arpingGratuitousARPOverIface   = arping.GratuitousARPOverIface
+	ndpGratuitousNDPOverIface      = ndp.GratuitousNDPOverIface
+	ipValidateExpectedInterfaceIPs = ip.ValidateExpectedInterfaceIPs
+	ipValidateExpectedRoute        = ip.ValidateExpectedRoute
+	ipGetVethPeerIfindex           = ip.GetVethPeerIfindex
+	getNSDevInterface              = util.GetNSDevInterface
+	getNSPeerDevBridge             = util.GetNSPeerDevBridge
+	nsGetNS                        = ns.GetNS
+	nsWithNetNSPath                = ns.WithNetNSPath
+	nsIsNSorErr                    = ns.IsNSorErr
+)
+
 type ifConfigurator struct {
 	ovsDatapathType             ovsconfig.OVSDatapathType
 	isOvsHardwareOffloadEnabled bool
 	disableTXChecksumOffload    bool
+	netlink                     netlinkutil.Interface
+	sriovnet                    SriovNet
 }
 
 func newInterfaceConfigurator(ovsDatapathType ovsconfig.OVSDatapathType, isOvsHardwareOffloadEnabled bool, disableTXChecksumOffload bool) (*ifConfigurator, error) {
-	return &ifConfigurator{ovsDatapathType: ovsDatapathType, isOvsHardwareOffloadEnabled: isOvsHardwareOffloadEnabled, disableTXChecksumOffload: disableTXChecksumOffload}, nil
+	configurator := &ifConfigurator{ovsDatapathType: ovsDatapathType, isOvsHardwareOffloadEnabled: isOvsHardwareOffloadEnabled, disableTXChecksumOffload: disableTXChecksumOffload, netlink: netlinkutil.NewClient()}
+	configurator.sriovnet = newSriovNet()
+	return configurator, nil
 }
 
-func renameLink(curName, newName string) error {
-	link, err := netlink.LinkByName(curName)
-	if err != nil {
-		return err
-	}
-	if err := netlink.LinkSetDown(link); err != nil {
-		return err
-	}
-	if err := netlink.LinkSetName(link, newName); err != nil {
-		return err
-	}
-	if err := netlink.LinkSetUp(link); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func moveIfToNetns(ifname string, netns ns.NetNS) error {
-	vfDev, err := netlink.LinkByName(ifname)
+func moveIfToNetns(netlinkClient netlinkutil.Interface, ifname string, netns ns.NetNS) error {
```

Comment:
could we make this func and moveVFtoContainerNS the member functions of ifConfigurator? then it doesn't need to pass the interface as an argument all the way down.

Reply:
I have a memory that some issue may exist ( not sure about it ) that  using a member function running inside a different netns. I would update it firstly, and run tests to see if failure happens.